karate-core/src/main/java/com/intuit/karate/Main.java

```diff
@@ -162,20 +169,52 @@ public static Main parseKarateOptions(String line) {
         return CommandLine.populateCommand(new Main(), args);
     }
 
-    public RuntimeHook createHook() {
-        if (hookFactoryClass != null) {
-            if (RuntimeHookFactory.class.isAssignableFrom(hookFactoryClass)) {
-                try {
-                    return ((RuntimeHookFactory) hookFactoryClass.newInstance()).create();
-                } catch (Exception e) {
-                    logger.error("Error instantiating RuntimeHook for {}", hookFactoryClass, e);
-                }
-            } else if (RuntimeHook.class.isAssignableFrom(hookFactoryClass)) {
-                try {
-                    return (RuntimeHook) hookFactoryClass.newInstance();
-                } catch (Exception e) {
-                    logger.error("Error instantiating RuntimeHook for {}", hookFactoryClass, e);
+    /**
+     * Parses Main options quoting last positional parameter (path) in case it contains whitespaces and it's unquoted.
+     * <p>
+     * It works only if line contains just one positional parameter (path) and it's the last one in line.
+     * It works well for IntelliJ and VSCode generated command lines.
+     *
+     * @param line
+     * @return
+     */
+    public static Main parseKarateOptionAndQuotePath(String line) {
```

Comment:
there is one utility created in `Command` here it is: 

https://github.com/intuit/karate/blob/develop/karate-core/src/main/java/com/intuit/karate/shell/Command.java#L123-L134

this is intended for parsing *any* command line into a string array, can you see if that is sufficient or improve it

then hopefully the existing logic to parse the last option a `path` should be sufficient

Reply:
ok, I think I understood Command.tokenize(), let me explain why I wrote it like this..
it's going to take me a few lines to put this in common

Karate and Picocli supports multiple options and parameters (whould be paths for karate) as long as they are quoted if they contain spaces

`-X option one.feature "with spaces.feature"`

so this would work ok if called using Main.main(String[] args) 

but if called from the IDEs it would came as a single String format, that is why Command.tokenize() should be used instead of string.split()

the problem is that IDEs (both intellij, vscode and probably eclipse) sends the feature path as an unquoted string even if it contains spaces :-(

until now the IdeMain class didn't make use of karateOptions, it just accepted a single string as the path (after sanitizing unknown options like --monochrome)
https://github.com/intuit/karate/blob/develop/karate-core/src/main/java/com/intuit/karate/cli/IdeMain.java#L103

Parsing a line like this `-H runtimeHook this feature has spaces.feature` using picocli would give you 1 option + 4 paths (instead of one) (Now it worked because it wasn't parsed for other options with picocli)

The function that I wrote, what it does is quoting last parameter and then it delegates to original IdeMain.parseKarateOptions that uses Command.tokenize()

It will produce `-H runtimeHook "this feature has spaces.feature"` that will split as expected in 3 strings by Command.tokenize()

(Caveant: `parseKarateOptionAndQuotePath` works only if there is one single path with no further options after path (it's the last one) which is compatible with what ides send.. And as I understand two unquoted paths with spaces would be an imposible task)


Now, **DapServer uses IdeMain.parseIdeCommandLine(launchCommand) but fas as I know debuging only works from VSCode**
https://github.com/intuit/karate/blob/develop/karate-core/src/main/java/com/intuit/karate/debug/DapServerHandler.java#L398

Does it makes more sense creating the quoting function **just to be used in this DapServerHandler** instead of inside IdeMain.parseIdeCommandLine **?**

But this will leave other IDEs out of the ability to configure RuntimeHooks on the command line (not even sure if this is posible)... 

just leaving it for a few days and sorry about this long story..

Reply:
thanks for the long explanation, it makes sense, I'll merge 👍